packages/calling-stateful-client/src/StreamUtils.ts

```diff
@@ -597,7 +597,12 @@ export function createView(
     return createViewLocalVideo(context, internalContext, callId, options);
   } else if (!('id' in stream) && !callId) {
     // Render LocalVideoStream that is not part of a Call
-    return createViewUnparentedVideo(context, internalContext, stream, options);
+    // Because it is not part of the call we don't tee errors to state naturally (e.g. via a Call Client function such as startVideo).
+    // We do not have a startLocalPreviewVideo function, so as a workaround we ensure any errors are propagated here.
+    return context.withAsyncErrorTeedToState(
```

Comment:
This is cool! So anytime we are passing around the CallContext in the adapter we can also kick the `withAsyncErrorTeedToState`? are there any situations in the chat composite/adapter that we should be thinking about this too?

Reply:
technically yes, but we should be careful when we call it as we don't want to trigger multiple times for the same error. ideally this is just wrapped around statefulclient handlers but since there isn't one for starting local preview video I added one here